NeoMathEngine/src/CPU/CpuMathEngineDnn3dConv.cpp

```diff
@@ -339,66 +330,46 @@ void CCpuMathEngine::blob3dConvolution( const CCommon3dConvolutionDesc& desc, co
 	const CBlobDesc& filter = desc.Filter;
 
 	const int objectCount = source.ObjectCount();
-
-	int preparedWidth = filter.GeometricalSize();
-
-	const int curThreadCount = IsOmpRelevant( objectCount * result.Width() * result.Depth(),
-		static_cast<int64_t>( source.BlobSize() ) * filter.BlobSize() ) ? threadCount : 1;
-	const int tempObjectCount = std::min( source.ObjectCount(), curThreadCount );
+	const int preparedWidth = filter.GeometricalSize();
+	const int tempObjectCount = source.ObjectCount();
```

Comment:
`tempObjectCount = std::min( source.ObjectCount(), curThreadCount );` without OMP means `tempObjectCount = 1;`.

`tempObjectCount = source.ObjectCount();` will lead to bigger memory consumption (and the code below still won't use more than 1 prepared object at a time so this allocation won't increase the performance)

Reply:
fixed